core/src/autogluon/core/models/abstract/abstract_model.py

```diff
@@ -564,6 +568,20 @@ def _fit(self,
         X = self.preprocess(X)
         self.model = self.model.fit(X, y)
 
+    def _apply_temperature_scaling(self, y_pred_proba):
+        if self.temperature_scalar is not None and self.problem_type in PROBLEM_TYPES_CLASSIFICATION:
```

Comment:
the fact temperature_scalar is set is enough to use it regardless of self.problem_type. If it is set when it is invalid, then that is the fault of the code that set it incorrectly in the first place. We can avoid the self.problem_type check here.

Reply:
removed `self.problem_type` check